addons/mass_mailing/models/mailing.py

```diff
@@ -518,7 +540,12 @@ def action_retry_failed(self):
         ])
         failed_mails.mapped('mailing_trace_ids').unlink()
         failed_mails.unlink()
-        self.write({'state': 'in_queue'})
+        mailing_send_immediate_nbr = self.env['ir.config_parameter'].sudo().get_param('mass_mailing.mass_mailing_immediate_send_nbr', 20)
```

Comment:
This could probably be extracted into its own sub-method by the way as it is present several times.